### PR TITLE
feat: Allow alternative country name spellings via array

### LINKED ISSUE #174: Allow for multiple possible spellings of names of a country
Currently, the localisation JSON maps strings 1-1. But in some languages, people may spell the country name differently.

I use this library to parse an API response into Alpha-2 codes. But, this external API sometimes uses slightly different spellings from those in this lib, but are completely valid spellings.

For instance, in Portuguese, mapping US to `Estados Unidos da América` is as valid as `Estados Unidos`.

Currently, I have to manually change the localisation file with different values, suited for my use-case or create a hard-coded matching algorithm.

I suggest we allow the possibility of accepting multiple names for a country in a given language. For instance, [langs/pt.json#L72](https://github.com/michaelwittig/node-i18n-iso-countries/blob/master/langs/pt.json#L72) would map to an array:

```diff
- "US": "Estados Unidos",
+ "US": ["Estados Unidos", "Estados Unidos da América"],
```
With this proposed syntax, if I run `countries.getName("US", "pt")`, the lib would return me the first entry in the array (`"Estados Unidos"`). But, when I do the opposite, either name would be valid.
```js
countries.getName("US", "pt") // Estados Unidos
countries.getAlpha2Code("Estados Unidos da América", "pt")) // US
countries.getAlpha2Code("Estados Unidos", "pt")) // US
```

or, to avoid confusion on array ordering, we could use a more complex object:
```diff
- "US": "Estados Unidos",
+ "US": {
+    "default": "Estados Unidos",
+    "alternatives": ["Estados Unidos da América"]
+ },
```

---

Checking the current implementation, it doesn't seem difficult to accomplish. [`getAlpha2Code`](https://github.com/michaelwittig/node-i18n-iso-countries/blob/master/index.js#L174) already searches the entire array of translations. It's just a matter of running 
```js
// TODO: Perform lowerCase on array or diacritics removal
Array.isArray(codenames[p]) ? codenames[p].includes(name) : codenames[p] === name
```

### PR DESCRIPTION
**Pull request by danilofuchs**
Closes #174

I added test cases and updated the translation maps for the example in the issue.

```js
countries.getName("US", "pt") // Estados Unidos
countries.getAlpha2Code("Estados Unidos da América", "pt")) // US
countries.getAlpha2Code("Estados Unidos", "pt")) // US
countries.getSimpleAlpha2Code("Estados Unidos da America", "pt")) // US
```

---

While building this, I found the code a little hard to follow, using `var` everywhere. I changed this bits to use `const` instead (supported since Node 6)

### CODE REVIEW

**Comment by michaelwittig** [comment]
LGTM

**Comment by michaelwittig** [comment]
released as `i18n-iso-countries@6.0.0`

**Comment by sebastien-prudhomme** [comment]
Hi,

Any reason to break compatibility for ```getNames``` function and not ```getName``` function?

I had to fix my code with this to have the same result than before: 

```js
Object.fromEntries(Object.entries(countries.getNames('en')).map(([key, value]) => [key, Array.isArray(value) ? value[0] : value]))
```

The new feature is good, but I would prefer to have a new function that get all known translations instead of changing the current one.

**Comment by danilofuchs** [comment]
> Hi,
> 
> Any reason to break compatibility for `getNames` function and not `getName` function?
> 
> I had to fix my code with this to have the same result than before:
> 
> ```js
> Object.fromEntries(Object.entries(countries.getNames('en')).map(([key, value]) => [key, Array.isArray(value) ? value[0] : value]))
> ```
> 
> The new feature is good, but I would prefer to have a new function that get all known translations instead of changing the current one.

I did not realise `getNames` interface would be changed, sorry about that. I think the user has more possibilities than before by having the names array.

Maybe we could add a `options.alternatives` optional argument in `getName` and `getNames` which either returns an array or the first match.

Either way, create a new issue so we can discuss it apart from this PR.
